src/raftstore/store/peer.rs

```diff
@@ -735,6 +738,14 @@ impl Peer {
                 self.notify_stale_command(cmd);
             }
 
+            if self.safe_conf_change {
+                if let Err(e) = self.check_conf_change(&req) {
```

Comment:
Check this in `propose_conf_change` can reuse the error handling code. And `propose_conf_change` should check if the request is a valid request too.